### PR TITLE
OTG Test - RT-2.1: Base IS-IS Process and Adjacencies

### LINKED ISSUE #589: OTG: RT-2.1: Base IS-IS Process and Adjacencies

### PR DESCRIPTION
**Pull request by octpetre**
OTG conversion of _**RT-2.1: Base IS-IS Process and Adjacencies**_
Fixes #589 
Output log [here](https://drive.google.com/file/d/1ts0jJ_xxZsOmBSUIsDE2T5fKoAXi667e/view?usp=share_link)

### CODE REVIEW

**Comment by octpetre** [comment]
@dplore could you please assign

File: feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go [hunk 1]
```diff
@@ -449,58 +454,103 @@ func TestTraffic(t *testing.T) {
...
 	dstIntf := ts.MustATEInterface(t, "port1")
+
+	srcIpv4 := srcIntf.Ethernets().Items()[0].Ipv4Addresses().Items()[0]
 	// net is a simulated network containing the addresses specified by targetNetwork
-	net := dstIntf.AddNetwork("net")
-	net.IPv4().WithAddress(targetNetwork.IPv4CIDR()).WithCount(1)
-	net.IPv6().WithAddress(targetNetwork.IPv6CIDR()).WithCount(1)
-	net.ISIS().WithIPReachabilityExternal().WithIPReachabilityMetric(10)
+	netv4 := dstIntf.Isis().V4Routes().Add().SetName("netv4").SetLinkMetric(10)
+	netv4.Addresses().Add().SetAddress(targetNetwork.IPv4)
```
Comment (jasdeep-hundal) [inline feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go:468]: Also add the prefix configuration here (even though that's the default, we should derive it from the targetNetwork struct to make the intent clear).

File: feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go [hunk 2]
```diff
@@ -449,58 +454,103 @@ func TestTraffic(t *testing.T) {
...
+	v4Flow.TxRx().Device().SetTxNames([]string{srcIpv4.Name()}).SetRxNames([]string{netv4.Name()})
+
+	v4FlowEth := v4Flow.Packet().Add().Ethernet()
+	v4FlowEth.Src().SetValue(session.ATETrafficAttrs.MAC)
+
+	v4FlowIp := v4Flow.Packet().Add().Ipv4()
+	v4FlowIp.Src().SetValue(session.ATETrafficAttrs.IPv4)
+	v4FlowIp.Dst().SetValue(targetNetwork.IPv4)
+
+	// v4Flow.Duration().FixedPackets().SetPackets(100)
```
Comment (jasdeep-hundal) [inline feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go:486]: Remove.

File: feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go [hunk 3]
```diff
@@ -449,58 +454,103 @@ func TestTraffic(t *testing.T) {
...
-		WithHeaders(ondatra.NewEthernetHeader(), v6Header)
-	// deadFlow is addressed to a nonexistent network as a consistency check -
-	// all traffic should be blackholed.
-	deadHeader := ondatra.NewIPv4Header()
-	deadHeader.DstAddressRange().WithMin(deadNetwork.IPv4).WithCount(1)
-	deadFlow := ate.Traffic().NewFlow("flow2").
-		WithSrcEndpoints(srcIntf).WithDstEndpoints(dstIntf).
-		WithHeaders(ondatra.NewEthernetHeader(), deadHeader)
+
+	time.Sleep(time.Second * 30)
```
Comment (jasdeep-hundal) [inline feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go:528]: Sleep statement got duplicated?

File: feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go [hunk 4]
```diff
@@ -449,58 +454,103 @@ func TestTraffic(t *testing.T) {
...
-		t.Errorf("Got %v%% IPv6 packet loss; expected < 1%%", v6Loss)
-	}
-	if deadLoss != 100 {
-		t.Errorf("Got %v%% invalid packet loss; expected 100%%", deadLoss)
+	otg.StopTraffic(t)
+
+	t.Logf("Verify Flow metrics...")
+	otgutils.LogFlowMetrics(t, otg, ts.ATETop)
+
+	for _, flow := range ts.ATETop.Flows().Items() {
```
Comment (jasdeep-hundal) [inline feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go:537]: Would prefer a helper that returns loss percentage and can be called individually for each flow. That makes it explicit which flows we are checking.

File: feature/experimental/isis/otg_tests/internal/session/session.go [hunk 5]
```diff
@@ -46,9 +47,11 @@ const (
 	DUTAreaAddress = "49.0001"
 	ATEAreaAddress = "49.0002"
 	DUTSysID       = "1920.0000.2001"
-	ISISName       = "DEFAULT"
-	pLen4          = 30
-	pLen6          = 126
+	ATESysID       = "640000000001"
+	// TODO: Change the name to DEFAULT
+	ISISName = "osisis"
```
Comment (jasdeep-hundal) [inline feature/experimental/isis/otg_tests/internal/session/session.go:52]: What's the reason for the change?

File: feature/experimental/isis/otg_tests/internal/session/session.go [hunk 6]
```diff
@@ -118,30 +123,43 @@ func addISISOC(dev *oc.Root, areaAddress, sysID, ifaceName string) {
...
+		SetAreaAddresses([]string{strings.Replace(areaAddress, ".", "", -1)})
+
+	devIsisInt := devIsis.Interfaces().
+		Add().
+		SetEthName(dev.Ethernets().Items()[0].Name()).
+		SetName("devIsisInt").
+		SetNetworkType(gosnappi.IsisInterfaceNetworkType.POINT_TO_POINT).
+		SetLevelType(gosnappi.IsisInterfaceLevelType.LEVEL_2)
+
+	devIsisInt.Advanced().
```
Comment (jasdeep-hundal) [inline feature/experimental/isis/otg_tests/internal/session/session.go:145]: Needed? (I think we removed this in another test)

File: feature/experimental/isis/otg_tests/base_adjacencies_test/README.md [hunk 7]
```diff
@@ -6,39 +6,22 @@ Base IS-IS functionality and adjacency establishment.
 
 ## Procedure
 
-*   Basic fields test
```
Comment (jasdeep-hundal) [inline feature/experimental/isis/otg_tests/base_adjacencies_test/README.md:9]: What's the reason for the doc changes? (AFAIK, I think they should capture the intent of the test, not necessarily the current state.)

**Review by jasdeep-hundal — CHANGES_REQUESTED** [review]

File: feature/experimental/isis/otg_tests/base_adjacencies_test/README.md (hunk 7 above)
Comment (jasdeep-hundal) [reply feature/experimental/isis/otg_tests/base_adjacencies_test/README.md:9]: @dplore : ^ Is my understanding correct there?

Comment (dplore) [reply feature/experimental/isis/otg_tests/base_adjacencies_test/README.md:9]: Yes, the readme should indicate the intent of the test.  If some of the intent is not implemented, we should note it in the readme with TODO.     

Looks like this proposes to remove coverage for hello padding?  (ie: only test with hello padding disabled?)

File: feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go [hunk 8]
```diff
@@ -449,58 +454,103 @@ func TestTraffic(t *testing.T) {
...
+	ts.ATEIntf1.Isis().Advanced().SetEnableHelloPadding(false)
 
-	ate := ts.ATE
+	// ate := ts.ATE
 	// We generate traffic entering along port2 and destined for port1
 	srcIntf := ts.MustATEInterface(t, "port2")
 	dstIntf := ts.MustATEInterface(t, "port1")
+
+	srcIpv4 := srcIntf.Ethernets().Items()[0].Ipv4Addresses().Items()[0]
 	// net is a simulated network containing the addresses specified by targetNetwork
```
[the remark this answers is not in the document]
Comment (octpetre) [reply feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go:466]: Done

File: feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go (hunk 1 above)
Comment (octpetre) [reply feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go:468]: The prefix configuration derived from the targetNetwork struct is done here ```netv4.Addresses().Add().SetAddress(targetNetwork.IPv4)``` for IPv4 and here ```netv6.Addresses().Add().SetAddress(targetNetwork.IPv6)``` for IPv6. Could you please clarify?

File: feature/experimental/isis/otg_tests/internal/session/session.go (hunk 5 above)
Comment (octpetre) [reply feature/experimental/isis/otg_tests/internal/session/session.go:52]: There is a problem with a vendor when the name is "DEFAULT". We're tracking the progress here https://github.com/openconfig/ka-fntest/issues/5

File: feature/experimental/isis/otg_tests/internal/session/session.go (hunk 6 above)
Comment (octpetre) [reply feature/experimental/isis/otg_tests/internal/session/session.go:145]: Removed

File: feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go (hunk 3 above)
Comment (octpetre) [reply feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go:528]: Replaced with an otg telemetry call to wait until the first prefix is received. Without this, the traffic won't start. In the future "PrefixAny()." will be replaced with "Prefix(targetNetwork.IPv4)" once this becomes available in otg

File: feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go (hunk 2 above)
Comment (octpetre) [reply feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go:486]: Removed

File: feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go [hunk 9]
```diff
@@ -406,20 +409,21 @@ func TestAuthentication(t *testing.T) {
...
+					ts.ATEIntf1.Isis().Interfaces().Items()[0].Authentication().SetAuthType("password")
+					ts.ATEIntf1.Isis().Interfaces().Items()[0].Authentication().SetPassword(password)
+				case oc.IsisTypes_AUTH_MODE_MD5:
+					ts.ATEIntf1.Isis().Interfaces().Items()[0].Authentication().SetAuthType("md5")
+					ts.ATEIntf1.Isis().Interfaces().Items()[0].Authentication().SetMd5(password)
+				default:
+					t.Fatalf("test case has bad mode: %v", tc.mode)
+				}
+			} else {
+				ts.ATEIntf1.Isis().RouterAuth().SetIgnoreReceiveMd5(true)
```
[the remark this answers is not in the document]
Comment (octpetre) [reply feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go:425]: Removed

File: feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go (hunk 3 above)
Comment (octpetre) [reply feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go:528]: Sleep was replaced with a otgtelemetry command to check for the incoming IS-IS L2 LSP. At some point the TODO section will be implemented as is more accurate.

File: feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go (hunk 1 above)
Comment (jasdeep-hundal) [reply feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go:461]: Ah, I mean that the prefix itself should be set explicitly:
netv4.Addresses().Add().SetAddress(targetNetwork.IPv4).SetPrefix(targetNetwork.IPv4Len)

File: feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go (hunk 4 above)
Comment (jasdeep-hundal) [reply feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go:541]: Remove the 'for' loop and just call it separately for each flow (since there are only 3 I think the readability is worth it to make explicit the check for each flow.)

File: feature/experimental/isis/otg_tests/base_adjacencies_test/README.md (hunk 7 above)
Comment (jasdeep-hundal) [reply feature/experimental/isis/otg_tests/base_adjacencies_test/README.md:9]: @octpetre : I think just add the TODO's to each section of the procedure with a note on the current deviation. (So keep the text largely the same.)

**Review by jasdeep-hundal — CHANGES_REQUESTED** [review]

File: feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go (hunk 4 above)
Comment (octpetre) [reply feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go:541]: Done

File: feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go (hunk 1 above)
Comment (octpetre) [reply feature/experimental/isis/otg_tests/base_adjacencies_test/base_adjacencies_test.go:462]: Added .SetPrefix(targetNetwork.IPv4Len) and .SetPrefix(int32(targetNetwork.IPv6Len)

File: feature/experimental/isis/otg_tests/base_adjacencies_test/README.md (hunk 7 above)
Comment (octpetre) [reply feature/experimental/isis/otg_tests/base_adjacencies_test/README.md]: Updated the README to match the ATE version

**Review by jasdeep-hundal — APPROVED** [review]
